Documentation/Proposals/NNNN-ranged-confirmations.md

```diff
@@ -0,0 +1,189 @@
+and [`PartialRangeThrough`](https://developer.apple.com/documentation/swift/partialrangethrough),
+may have surprising behavior when used with this new interface because they
+implicitly include `0`. If a test author writes `...10`, do they mean "zero to
+ten" or "one to ten"? The programmatic meaning is the former, but some test
+authors might mean the latter. If an event does not occur, a test using
+`confirmation()` and this `expectedCount` value would pass when the test author
+meant for it to fail.
+
+The unbounded range (`...`) type `UnboundedRange` is effectively useless when
+used with this interface and any use of it here is almost certainly a programmer
+error.
+
+`PartialRangeUpTo` and `PartialRangeThrough` conform to `RangeExpression`, but
+not to `Sequence`, so they will be rejected at compile time. `UnboundedRange` is
+a non-nominal type and will not match either. We will provide unavailable
+overloads of `confirmation()` for these types with messages that explain why
+they are unavailable, e.g.:
+
+```swift
+@available(*, unavailable, message: "Unbounded range '...' has no effect when used with a confirmation.")
+public func confirmation<R>(
+  _ comment: Comment? = nil,
+  expectedCount: UnboundedRange,
+  isolation: isolated (any Actor)? = #isolation,
+  sourceLocation: SourceLocation = #_sourceLocation,
+  _ body: (Confirmation) async throws -> R
+) async rethrows -> R
+```
+
+## Source compatibility
+
+This change is additive. Existing tests are unaffected.
+
+Code that refers to `confirmation(_:expectedCount:isolation:sourceLocation:_:)`
+by symbol name may need to add a contextual type to disambiguate the two
+overloads at compile time.
+
+## Integration with supporting tools
+
+The type of the associated value `expected` for the `Issue.Kind` case
+`confirmationMiscounted(actual:expected:)` will change from `Int` to
+`any RangeExpression & Sendable`[^1]. Tools that implement event handlers and
+distinguish between `Issue.Kind` cases are advised not to assume the type of
+this value is `Int`.
+
+## Alternatives considered
+
+- Doing nothing. We have identified real-world use cases for this interface
+  including in Swift Testing’s own test target.
+- Allowing the use of any value as the `expectedCount` argument so long as it
+  conforms to a protocol `ExpectedCount` (we'd have range types and `Int`
+  conform by default.) It was unclear what this sort of flexibility would let
+  us do, and posed challenges for encoding and decoding events and issues when
+  using the JSON event stream interface.
+
+## Acknowledgments
+
+If significant changes or improvements suggested by members of the community
+were incorporated into the proposal as it developed, take a moment here to thank
+them for their contributions. This is a collaborative process, and everyone's
+input should receive recognition!
+
+Generally, you should not acknowledge anyone who is listed as a co-author.
```

Comment:
Did you intend to leave the acknowledgements section in?

Reply:
Oops. Will correct.